lib/routes/user/home/payments_filter.dart

```diff
@@ -151,4 +153,54 @@ class PaymentsFilterState extends State<PaymentsFilter> {
     }
     return [PaymentType.RECEIVED, PaymentType.DEPOSIT, PaymentType.SENT, PaymentType.WITHDRAWAL];
   }
+
+  IconButton _buildExportButton(BuildContext context) {
+    return widget._paymentsModel.paymentsList.isNotEmpty
+        ? IconButton(
+            icon: Icon(
+              Icons.share,
+              color: Colors.white,
+              size: 18.0,
+            ),
+            onPressed: () {
+              List<List<dynamic>> paymentListArr = new List.generate(widget._paymentsModel.paymentsList.length, (index) {
```

Comment:
Better to use an async action (ExportPayments) and do the export business logic in the bloc.
The action may return the file and the UI only shares it.

Reply:
Done 👍 (but we should refactor certain parts)